src/interface/extension_interface.hpp

```diff
@@ -596,6 +597,54 @@ typename sb_handle_t::event_t _reduction(
         sb_handle, buffer_in, ld, buffer_out, rows, cols, dependencies);
   }
 }
+template <typename sb_handle_t, typename container_0_t, typename container_1_t,
+          typename element_t, typename index_t>
+typename sb_handle_t::event_t _axpy_batch(
+    sb_handle_t& sb_handle, index_t _N, element_t _alpha, container_0_t _vx,
+    index_t _incx, index_t _stride_x, container_1_t _vy, index_t _incy,
+    index_t _stride_y, index_t _batch_size,
+    const typename sb_handle_t::event_t& _dependencies) {
+  return blas::axpy_batch::backend::_axpy_batch(
+      sb_handle, _N, _alpha, _vx, _incx, _stride_x, _vy, _incy, _stride_y,
+      _batch_size, _dependencies);
+}
+
+template <int localSize, int maxBlockPerBatch, typename sb_handle_t,
+          typename container_0_t, typename container_1_t, typename element_t,
+          typename index_t>
+typename sb_handle_t::event_t _axpy_batch_impl(
+    sb_handle_t& sb_handle, index_t _N, element_t _alpha, container_0_t _vx,
+    index_t _incx, index_t _stride_x, container_1_t _vy, index_t _incy,
+    index_t _stride_y, index_t _batch_size,
+    const typename sb_handle_t::event_t& _dependencies, index_t global_size) {
+  // if inc are of opposite sign the values are exchanged. It doesn't matter
+  // which one is positive or negative, so to simplify index computation in
+  // kernel we always set incx to be negative and incy to be positive.
+  if (_incx > 0 && _incy < 0) {
+    _incx = -_incx;
+    _incy = -_incy;
+  }
+  typename VectorViewType<container_0_t, index_t, index_t>::type vx =
+      make_vector_view(_vx, static_cast<index_t>(_incx),
+                       static_cast<index_t>(_N * _batch_size));
+  auto vy = make_vector_view(_vy, _incy, _N * _batch_size);
```

Comment:
You are right that was not correct. The correct computation is `_stride_x/y * _batch_size` since the size of the input array is given by all batch and the distance between them. Thank you for spotting it, it pointed me in the right direction. I addressed it in 0ccfe3e5c2f0cfc85ef2792678d3f792e600da5e